s2/builder_graph_options.go

```diff
@@ -0,0 +1,179 @@
+// choosing arbitrarily, instead we merge the labels of all duplicate edges
+// (even ones where no sibling pairs were discarded), yielding {AB123,
+// AB123, CD45, CD45} (assuming that duplicate edges are being kept).
+// Notice that the labels of duplicate edges are merged even if no siblings
+// were discarded (such as CD5, CD6 in this example), and that this would
+// happen even with duplicate degenerate edges (e.g. the edges EE7, EE8).
+type siblingPairs uint8
+
+const (
+	// siblingPairsDiscard discards all sibling edge pairs.
+	siblingPairsDiscard siblingPairs = iota
+	// siblingPairsDiscardExcess is like siblingPairsDiscard, except that a
+	// single sibling pair is kept if the result would otherwise be empty.
+	// This is useful for polygons with degeneracies (LaxPolygon), and for
+	// simplifying polylines while ensuring that they are not split into
+	// multiple disconnected pieces.
+	siblingPairsDiscardExcess
+	// siblingPairsKeep keeps sibling pairs. This can be used to create
+	// polylines that double back on themselves, or degenerate loops (with
+	// a layer type such as LaxPolygon).
+	siblingPairsKeep
+	// siblingPairsRequire requires that all edges have a sibling (and returns
+	// an error otherwise). This is useful with layer types that create a
+	// collection of adjacent polygons (a polygon mesh).
+	siblingPairsRequire
+	// siblingPairsCreate ensures that all edges have a sibling edge by
+	// creating them if necessary. This is useful with polygon meshes where
+	// the input polygons do not cover the entire sphere. Such edges always
+	// have an empty set of labels and do not have an associated InputEdgeID.
+	siblingPairsCreate
+)
+
+// graphOptions is only needed by Layer implementations. A layer is
+// responsible for assembling an Graph of snapped edges into the
+// desired output format (e.g., an Polygon). The graphOptions allows
+// each Layer type to specify requirements on its input graph: for example, if
+// degenerateEdgesDiscard is specified, then Builder will ensure that all
+// degenerate edges are removed before passing the graph to Layer's Build
+// method.
+type graphOptions struct {
+	// Specifies whether the Builder input edges should be treated as
+	// undirected. If true, then all input edges are duplicated into pairs
+	// consisting of an edge and a sibling (reverse) edge. Note that the
+	// automatically created sibling edge has an empty set of labels and does
+	// not have an associated InputEdgeId.
+	//
+	// The layer implementation is responsible for ensuring that exactly one
+	// edge from each pair is used in the output, i.e. *only half* of the graph
+	// edges will be used. (Note that some values of the siblingPairs option
+	// automatically take care of this issue by removing half of the edges and
+	// changing edgeType to Directed.)
+	//
+	// DEFAULT: edgeTypeDirected
+	edgeType edgeType
+
+	// DEFAULT: degenerateEdgesKeep
```

Comment:
Could we make the default values the first iota value when declaring the constants?

Reply:
I've reordered all the builder enums to have the first iota value be the expected default option.